Support for REDIS ZSET

amaury-d opened the pull request:
This commit adds support for ZSET queues in REDIS with data_type `sortedset`.
Batch is supported.

Priority is based on a field of the message containing a number.
If the field is absent or not a number a default priority is then used.

It adds the following options to the configuration:
* **priority_field**: *string*, default 'epoch', Priority field to
use, if field doesn't exist, priority will be priority_default
* **priority_default**: *number*, default '-1', Default priority when
priority field is not found in the event

Usage example:
 ```
output {
    redis {
		host => ["127.0.0.1"]
		port => 6379
		data_type => "sortedset"
		key => "syslog"
		priority_field => "myepochfield"
    }
 }
```
=> The messages will be push to a sortedset with the key `syslog` using the content of the field `myepochfield` as score.


Thanks for contributing to Logstash! If you haven't already signed our CLA, here's a handy link: https://www.elastic.co/contributor-agreement/

amaury-d committed:
Support for REDIS priority queues

This commit adds support for ZSET queues in REDIS with data_type 'list'.
Batch is supported.

Priority is based on a field of the message containing a number.
If the field is absent or not a number a default priority is then used.

It adds the following options to the configuration:
* **priority**: *boolean*, default false, enable priority mode
* **priority_field**: *string*, default '@timestamp', Priority field to
use, if field doesn't exist, priority will be priority_default
* **priority_default**: *number*, default '0', Default priority when
priority field is not found in the event

Usage example:
    output {
    	redis {
		host => ["127.0.0.1"]
		port => 6379
		data_type => "list"
		key => "syslog"
		priority => true
		priority_field => "epoch"
	}
    }

amaury-d committed:
Support for REDIS priority queues

This commit adds support for ZSET queues in REDIS with data_type 'list'.
Batch is supported.

Priority is based on a field of the message containing a number.
If the field is absent or not a number a default priority is then used.

It adds the following options to the configuration:
* **priority**: *boolean*, default false, enable priority mode
* **priority_field**: *string*, default '@timestamp', Priority field to
use, if field doesn't exist, priority will be priority_default
* **priority_default**: *number*, default '0', Default priority when
priority field is not found in the event

Usage example:
    output {
    	redis {
		host => ["127.0.0.1"]
		port => 6379
		data_type => "list"
		key => "syslog"
		priority => true
		priority_field => "epoch"
	}
    }

amaury-d commented:
This pull request is linked to https://github.com/logstash-plugins/logstash-input-redis/pull/56

untergeek commented:
@amaury-d Can you please add some rspec tests for these changes? This is a rather significant update and some tests are absolutely necessary.

amaury-d commented:
@untergeek Sure, i'll work on it and update the pull request

amaury-d committed:
Added batch_timeout to redis batch unit testing

Sometimes redis didn't have the time to ingest all messages and the
test fails randomly.
It appeared it was due to redis too short timeout.

amaury-d committed:
Added batch_timeout to redis batch unit testing

Sometimes redis didn't have the time to ingest all messages and the
test fails randomly.
It appeared it was due to redis too short timeout.

amaury-d committed:
Sorted set: Spec and support sortedset data_type

As sorted sets are a native redis data type, it's best to support ZSET
through data_type option instead to boolean priority option.

A first group of integration tests (batched and unbatched) for
sorted sets is also added with this commit.

amaury-d committed:
Sorted set: Spec and support sortedset data_type

As sorted sets are a native redis data type, it's best to support ZSET
through data_type option instead to boolean priority option.

A first group of integration tests (batched and unbatched) for
sorted sets is also added with this commit.

amaury-d committed:
ZADD integration spec

amaury-d committed:
ZADD integration spec

amaury-d committed:
Priority in sortedset is a string

amaury-d committed:
Priority in sortedset is a string

amaury-d committed:
Sorted test integration test non batch ok

amaury-d committed:
Sorted test integration test non batch ok

amaury-d committed:
Sortedset batch integration testing done

amaury-d committed:
Sortedset batch integration testing done

amaury-d committed:
Merge branch 'master' of https://github.com/logstash-plugins/logstash-output-redis

amaury-d committed:
Merge branch 'master' of https://github.com/amaury-d/logstash-output-redis

amaury-d committed:
Merge branch 'master' of https://github.com/logstash-plugins/logstash-output-redis

amaury-d committed:
Documentation for data_type `sortedset`

amaury-d committed:
Typo

amaury-d commented:
Hi @untergeek @andrewvc ,

 rspec tests and supplementary documentation have been added to pull requests (input and output).

zivziv77 commented:
Hello,

How can I download you changes? I need them also

Thanks

amaury-d commented:
@zivziv77 just go on my github or check the diff of that pull request

zivziv77 commented:
Thanks,

why not putting it in master and have it in the official release?

zivziv77 commented:
Would you be nice and explain to me step step how can I install it on prod
machine?

On Tue, 26 Jun 2018 at 19:19 Amaury Decreme <notifications@github.com>
wrote:

> @zivziv77 <https://github.com/zivziv77> just go on my github or check the
> diff of that pull request
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/logstash-plugins/logstash-output-redis/pull/46#issuecomment-400373995>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALyLUPXbM9vdPMckMhGMLfkgvC5A4Cfpks5uAl8XgaJpZM4L5bFX>
> .
>

mudit1804 commented:
Does it supports NX/XX options along with ZADD command ?
